src/shogun/machine/RRForest.cpp

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) 2016, Shogun-Toolbox e.V. <shogun-team@shogun-toolbox.org>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  1. Redistributions of source code must retain the above copyright notice,
+ *     this list of conditions and the following disclaimer.
+ *
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *
+ *  3. Neither the name of the copyright holder nor the names of its
+ *     contributors may be used to endorse or promote products derived from
+ *     this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Written (W) 2016 Saurabh Mahindre
+ */
+
+#include <shogun/machine/RRForest.h>
+#include <shogun/multiclass/tree/RRCARTree.h>
+
+using namespace shogun;
+
+CRRForest::CRRForest()
+: CBaggingMachine()
+{
+	init();
+}
+
+CRRForest::CRRForest(int32_t rand_numfeats, int32_t num_bags)
+: CBaggingMachine()
+{
+	init();
+	
+	set_num_bags(num_bags);
+
+	if (rand_numfeats>0)
+		dynamic_cast<CRRCARTree*>(m_machine)->set_feature_subset_size(rand_numfeats);
+}
+
+CRRForest::CRRForest(CFeatures* features, CLabels* labels, int32_t num_bags, int32_t rand_numfeats)
+: CBaggingMachine()
+{
+	init();
+
+	SG_REF(features);
+	m_features=features;
+	set_labels(labels);
+
+	set_num_bags(num_bags);
+
+	if (rand_numfeats>0)
+		dynamic_cast<CRRCARTree*>(m_machine)->set_feature_subset_size(rand_numfeats);
+}
+
+CRRForest::CRRForest(CFeatures* features, CLabels* labels, SGVector<float64_t> weights, int32_t num_bags, int32_t rand_numfeats)
```

Comment:
rand_num_feats
Maybe a better name is appropriate as well: num_random_feats, num_rand_feats, or so